examples/stable-diffusion-jax/alpa_run.py

```diff
@@ -0,0 +1,145 @@
+import optax
+
+alpa.init(cluster="ray")
+#logger = logging.getLogger(__name__)
+
+def image_grid(imgs, rows, cols):
+    assert len(imgs) == rows*cols
+
+    w, h = imgs[0].size
+    grid = Image.new('RGB', size=(cols*w, rows*h))
+    grid_w, grid_h = grid.size
+    
+    for i, img in enumerate(imgs):
+        grid.paste(img, box=(i%cols*w, i//cols*h))
+    return grid
+
+
+# convert diffusers checkpoint to jax
+#pt_path = "/home/patrick/stable-diffusion-v1-3"
+#fx_path = pt_path + "_jax"
+#convert_diffusers_to_jax(pt_path, fx_path)
+
+fx_path = "/data/wly/stable-diffusion-jax/local_hf/sd-v1-4-flax"
+
+# inference with jax
+dtype = jnp.float16
+clip_model, clip_params = FlaxCLIPTextModel.from_pretrained(
+    "openai/clip-vit-large-patch14", _do_init=False, dtype=dtype
+)
+unet, unet_params = UNet2D.from_pretrained(f"{fx_path}/unet", _do_init=False, dtype=dtype)
+vae, vae_params = AutoencoderKL.from_pretrained(f"{fx_path}/vae", _do_init=False, dtype=dtype)
+safety_checker, safety_params = StableDiffusionSafetyCheckerModel.from_pretrained(f"{fx_path}/safety_checker", _do_init=False)
+scheduler = PNDMScheduler.from_config(f"{fx_path}/scheduler", use_auth_token=True)
+feature_extractor = AutoFeatureExtractor.from_pretrained("openai/clip-vit-large-patch14")
+
+tokenizer = CLIPTokenizer.from_pretrained("openai/clip-vit-large-patch14")
+
+# create inference state and replicate it across all TPU devices
+inference_state = InferenceState(text_encoder_params=clip_params, unet_params=unet_params, vae_params=vae_params)
+
+
+# inference_state = replicate(inference_state)
+
+
+# create pipeline
+pipe = StableDiffusionPipeline(text_encoder=clip_model, tokenizer=tokenizer, unet=unet, scheduler=scheduler, vae=vae)
+
+
+# prepare inputs
+num_samples = 8
+#p = "A cinematic film still of Morgan Freeman starring as Jimi Hendrix, portrait, 40mm lens, shallow depth of field, close up, split lighting, cinematic"
+#p = "a photograph of an astronaut riding a horse"
+p = "In a Cyberpunk city, cyberpunk knights ride on cyberpunk horses"
+
+input_ids = tokenizer(
+    [p] * num_samples, padding="max_length", truncation=True, max_length=77, return_tensors="jax"
+).input_ids
+uncond_input_ids = tokenizer(
+    [""] * num_samples, padding="max_length", truncation=True, max_length=77, return_tensors="jax"
+).input_ids
+prng_seed = jax.random.PRNGKey(1)
+
+# shard inputs and rng
+#input_ids = shard(input_ids)
+#uncond_input_ids = shard(uncond_input_ids)
+#prng_seed = jax.random.split(prng_seed, 2)
+
+
+# pmap the sample function
+num_inference_steps = 1
+guidance_scale = 7.5
+
+
+#sample = jax.pmap(pipe.sample, static_broadcasted_argnums=(4, 5))
+sample = pipe.sample
+
+
+def inf_step(
+    input_ids,
+    uncond_input_ids,
+    inference_state
+):
+    return pipe.sample(
+    input_ids,
+    uncond_input_ids,
+    prng_seed,
+    inference_state,
+    num_inference_steps,
+    guidance_scale,
+)
+
+inf_step = alpa.parallelize(inf_step, method=alpa.DataParallel())
```

Comment:
both the `input_ids` and `uncond_input_ids` are variables with a batch dimension, please set the [`batch_argnums`](https://github.com/alpa-projects/alpa/blob/bf62c1763c0fc3ec58e824ce421a010728af5ae5/alpa/api.py#L67) correctly.